python/examples/explore.py

```diff
@@ -0,0 +1,295 @@
+# load the structures from the dataset. Here we use an `ASE package
+# <https://wiki.fysik.dtu.dk/ase/>`_ to read the structures from the file and have the
+# frames as the `ase.Atoms <https://wiki.fysik.dtu.dk/ase/ase/atoms.html/>`_ objects.
+
+frames = ase.io.read("data/explore_c-gap-20u.xyz", ":")
+
+
+# %%
+#
+# Provide the frames to the `chemiscope.explore`. It will generate a Chemiscope
+# interactive widget with the reduced dimentionality of data.
+
+chemiscope.explore(frames)
+
+# %%
+#
+# In this basic case, no featurizer function is provided, so `chemiscope.explore` uses a
+# default method that applies `SOAP (Smooth Overlap of Atomic Positions)
+# <https://singroup.github.io/dscribe/latest/tutorials/descriptors/soap.html/>`_ to
+# compute atomic structure descriptors and then performs `PCA (Principal Component
+# Analysis)
+# <https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html/>`_
+# for dimensionality reduction. The resulting components are then added to the
+# properties to be used in visualization.
+
+
+# %%
+#
+# Example with SOAP and KPCA
+# ++++++++++++++++++++++++++
+#
+# This part illustrates how to create a custom function for dimensionality reduction
+# as an argument (`featurize`) to `chemiscope.explore`. Inside this function, we perform
+# descriptor calculation using SOAP and then reduce the dimensionality with `Kernel PCA
+# <https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.KernelPCA.html/>`_.
+#
+# First, let's import the necessary packages.
+from dscribe.descriptors import SOAP  # noqa
+from sklearn.decomposition import KernelPCA  # noqa
+
+# %%
+#
+# Define the function `soap_kpca` which takes one argument (`frames`). This argument
+# contains the structures provided to `chemiscope.explore` and is internally passed
+# to the `featurize` function.
+
+
+def soap_kpca(frames):
+    # Initialise soap calculator. The detailed explanation of the provided
+    # hyperparameters can be checked in the documentation of the library (`dscribe`).
+    soap = SOAP(
+        # the dataset used in the example contains only carbon
+        species=["C"],
+        r_cut=4.5,
+        n_max=8,
+        l_max=6,
+        sigma=0.2,
+        rbf="gto",
+        average="outer",
+        periodic=True,
+        weighting={"function": "pow", "c": 1, "m": 5, "d": 1, "r0": 3.5},
+    )
+
+    # Compute features
+    descriptors = soap.create(frames)
+
+    # Apply KPCA
+    transformer = KernelPCA(n_components=2, gamma=0.05)
+
+    # Return a 2D array of reduced features
+    return transformer.fit_transform(descriptors)
+
+
+# %%
+#
+# Provide the created function to `chemiscope.explore`.
+
+cs = chemiscope.explore(frames, featurize=soap_kpca)
+
+# %%
+#
+# Note: It is possible to add parallelization when computing the SOAP descriptors and
+# performing dimentionality reduction with KernelPCA by providing the `n_jobs`
```

Comment:
does KernelPCA have a `n_jobs` parameter? I though this was only for dscribe

Reply:
Yes, KernelPCA also has this parameter
https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.KernelPCA.html